### ISSUE TITLE
Not all handler functions must necessarily return a promise

### ISSUE THREAD

**Issue by FetzenRndy** [AndyGura/nestjs-socket-handlers-with-ack#4]
## The Problem

In your decorator you bind the original function and then call "then" ([here](https://github.com/AndyGura/nestjs-socket-handlers-with-ack/blob/d578c029a83acfdbe095da1a551049a6a6aa592b/src/decorators/subscribe-message-with-ack.decorator.ts#L12))

While this is fine when the handler is async or returns a promise, a handler is not necessarily async.
When you have a sync handler like this:

##### Notice the lack of the "async" keyword
```ts
    @SubscribeMessageWithAck('checkIsNumberOdd')
    checkIsNumberOdd(socket: SocketIO.Socket, num: number): boolean {
        if (isNaN(num)) {
            throw new Error('Wrong number received');
        }
        return num % 2 === 1;
    }
```

Your code will not work.

## How to fix?

You should first check if the handler is a promise, and if it is invoked it like you do, and if it not, invoke it
synchronously

If you want me to send a Pull request let me know

## Relevant resources

https://stackoverflow.com/questions/27746304/how-do-i-tell-if-an-object-is-a-promise

**Comment by AndyGura** [comment]
Fixed in 633d80ad8824faebd06d216e6bf89c5aa6e8d69c

**Comment by AndyGura** [comment]
Fixed in version 1.1.0
